### PR TITLE
add name or private directory being used to the window title when xpra is being used

### PR DESCRIPTION
**Pull request by dmfreemon**
First, thanks for making firejail available.  It is outstanding.

I use a number of different private directories, and wanted some kind of visual indication of which private directory was being used by each instance of a running firefox.  This PR adds this information to the window title when xpra is being used.  I find this to be very valuable for me, and maybe others will as well.

This PR may be related to at least some of this prior issues (in no particular order):
https://github.com/netblue30/firejail/issues/1983
https://github.com/netblue30/firejail/issues/1350
https://github.com/netblue30/firejail/issues/1308
https://github.com/netblue30/firejail/issues/281
https://github.com/netblue30/firejail/issues/303
https://github.com/netblue30/firejail/issues/424
https://github.com/netblue30/firejail/issues/242

### CODE REVIEW

File: src/firejail/x11.c [hunk 1]
```diff
@@ -625,6 +625,52 @@ void x11_start_xephyr(int argc, char **argv) {
...
+
+	char * title_arg_str = NULL;
+
+	const char * title_start = "--title=firejail x11 sandbox";
+	const char * title_sep = " ";
+
+	// use the "--name" argument if it was explicitly specified
+	if ((cfg.name != NULL) && (strlen(cfg.name) > 0)) {
+
+		title_arg_str = malloc(strlen(title_start) + strlen(title_sep) + strlen(cfg.name) + 1);
```
Comment (smitsohu) [inline src/firejail/x11.c]: Could you check the return values of the malloc calls?

File: src/firejail/x11.c [hunk 2]
```diff
@@ -625,6 +625,52 @@ void x11_start_xephyr(int argc, char **argv) {
...
+		strcpy(title_arg_str, title_start);
+		strcat(title_arg_str, title_sep);
+		strcat(title_arg_str, cfg.name);
+	}
+
+	// use the "--private" argument if it was explicitly specified
+	else if ((cfg.home_private != NULL) && (strlen(cfg.home_private) > 0)) {
+
+		char * tmp_in = strdupa(cfg.home_private);
+		char * base_out = strdupa(basename(tmp_in));
```
Comment (smitsohu) [inline src/firejail/x11.c]: Did you consider using the `gnu_basename` we carry in util.c? It doesn't modify the argument

File: src/firejail/x11.c (hunk 1 above)
Comment (dmfreemon) [reply src/firejail/x11.c]: Done.  Commit 7f39af86.

File: src/firejail/x11.c (hunk 2 above)
Comment (dmfreemon) [reply src/firejail/x11.c]: Done.  Commit 7f39af86.

**Comment by smitsohu** [comment]
Merged, thanks!

**Pull request merged**